rasa/core/tracker_store.py

```diff
@@ -289,13 +289,19 @@ async def retrieve_full_tracker(
 
     async def stream_events(self, tracker: DialogueStateTracker) -> None:
         """Streams events to a message broker."""
+        offset = await self.number_of_existing_events(tracker.sender_id)
```

Comment:
I'd suggest still doing the `self.event_broker is None` check at the very beginning here like before, instead of in `stream_new_events()`.

Reply:
Oh I moved that because mypy check would fail with a `union-attr` error type and it didn't make sense to have this duplicated in 2 places: 
```
rasa/core/tracker_store.py:306: error: Item "None" of "Optional[EventBroker]" has no attribute "publish"  [union-attr]
```

Reply:
Then one way to keep the original flow would be to refactor this so we pass the broker as an argument in `stream_new_events()` instead of accessing the instance field - then we can do the check at the beginning but mypy should still be happy that way.

Reply:
I'll try that 👍🏼

Reply:
Seems I cannot pass `self.event_broker` as argument, therefore I have to assign this attribute or copy it to a different local variable, something in this seems wrong although cannot say for sure why 🤔

Reply:
what's the error when you try passing it as an argument?

Reply:
Interesting, my pycharm seems perfectly happy with this solution. But in any case I'd say our linter should be the authoritative source on what's allowed and what's not 🙂 static method works too of course 👍